include/tlog/journal_json_writer.h

```diff
@@ -68,9 +68,9 @@ tlog_journal_json_writer_create(struct tlog_json_writer **pwriter,
 {
     assert(pwriter != NULL);
     assert(tlog_syslog_priority_is_valid(priority));
-    assert(!augment || recording != NULL);
-    assert(!augment || username != NULL);
-    assert(!augment || session_id != 0);
```

Comment:
These `assert`s basically verify that the caller either didn't ask for augmentation, or if they asked, that they provided the required data to put into the extra fields. This has only to do with verifying the caller is sane. It has nothing to do with whether `augment` is configurable or not. This code doesn't know anything about that and it should not. It should only take care that the parameters it's provided make sense. It will then be the job of the configuration-handling code to verify that no conflicting configuration is possible, which would trip these `assert`s, or, if they're disabled, would make the code crash (because an argument was NULL). In short, separation of concern.

Reply:
Thanks for the explanation! I will close this one out.